### ISSUE TITLE
Connecting to Kafka triggers DEP0123

### ISSUE THREAD

**Issue by milibopp** [tulios/kafkajs#1354]
**Describe the bug**
When using a broker identified by a domain name, its IP addresses seems to be used for domain resolution rather than its IP. The domain name appears to resolve to multiple IP addresses. (I'm guessing those are the different Kafka instances of the cluster.)

**To Reproduce**
Connect to a Kafka cluster identified by a domain name that refers to multiple IP addresses. (In my case a Kafka cluster hosted by [Aiven](https://aiven.io/) triggered the issue.)

**Expected behavior**
No warnings.

**Observed behavior**
```
(node:40285) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
```

**Environment:**
 - OS: NixOS 21.11
 - KafkaJS version 1.16.0
 - Kafka version 3.1
 - NodeJS version 16.14.2

**Comment by Nevon** [comment]
Can you reproduce this with [2.2.0-beta.2](https://www.npmjs.com/package/kafkajs/v/2.2.0-beta.2)? #1425 made it so that we don't set the `servername` by default if the host is an IP address rather than a hostname.

My guess, without having any access to aiven, is that the issue is not with the first broker(s) that you add to the config, but rather the subsequent brokers that are discovered. I'm guessing that the domain just resolves to one of the brokers, and the TLS ServerName will be the domain you provided, but that broker then returns a list of IPs rather than domain names, so subsequent connections use the IP as ServerName, triggering this warning.

**Comment by milibopp** [comment]
Ah, yes, I checked application logs: the warning appeared upon daily server restart on July 20th for the last time. No version upgrade of KafkaJS was involved, so I would guess something about DNS or another aspect of the cluster infrastructure must have changed on Aiven's side at that point, so that our setup no longer reproduces this issue.

**Comment by Nevon** [comment]
Alright, then I will close this, as this _should_ have been fixed by #1425 regardless.
